drivers/serial/uart_ns16550.c

```diff
@@ -467,6 +470,8 @@ static int uart_ns16550_init(struct device *dev)
 	DEV_CFG(dev)->devconf.irq_config_func(dev);
 #endif
 
+	DEV_DATA(dev)->port = DEV_CFG(dev)->devconf.port;
```

Comment:
When PCIE is enabled, `port` is already set inside `uart_ns16550_configure()`. Setting this here will override it.

I think it is probably better to move the the whole `pcie_probe()` part here before calling `uart_ns16550_configure()`. The port only needs to be probed once, and there is no need to probe PCIE every single time UART is being configured.

Reply:
This is probably why it didn't work on the UP2 board.

Reply:
@dcpleung Thank you! That's a great point. A note - if I remove this line 473, the applications which do not use PCIE do not get configured. So, I can specify #ifndef PCIE for this line, and I'll move the whole pcie_probe() here before calling _configure(). I'll push an update shortly!